src/plugins/intel_cpu/src/transformations/snippets/normalize_convert_pre_lower.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "transformations/snippets/normalize_convert_pre_lower.hpp"
+
+#include <openvino/op/convert.hpp>
+#include <openvino/opsets/opset1.hpp>
+#include <openvino/pass/manager.hpp>
+#include <openvino/pass/pattern/matcher.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+
+#include "snippets/op/convert_truncation.hpp"
+#include "snippets/op/subgraph.hpp"
+
+namespace ov::intel_cpu::pass {
+
+namespace {
+class ReplaceConvertInBody : public ov::pass::MatcherPass {
+public:
+    OPENVINO_MATCHER_PASS_RTTI("CPU::ReplaceConvertInBody");
+    ReplaceConvertInBody() {
+        auto m_v0 = ov::pass::pattern::wrap_type<ov::op::v0::Convert>();
+        auto m_v1 = ov::pass::pattern::wrap_type<ov::opset1::Convert>();
+
+        auto callback = [](ov::pass::pattern::Matcher& m) {
+            const auto root = m.get_match_root();
+            std::shared_ptr<ov::Node> cv = root;
+            ov::element::Type dst;
+            if (auto c0 = std::dynamic_pointer_cast<ov::op::v0::Convert>(cv)) {
+                dst = c0->get_destination_type();
+            } else if (auto c1 = std::dynamic_pointer_cast<ov::opset1::Convert>(cv)) {
+                dst = c1->get_destination_type();
+            } else {
+                return false;
+            }
```

Comment:
dst can be always get as `cv->get_output_element_type(0)`, we can avoid the casting here